### PR TITLE
Add `audience` to the `client_options`

### PR DESCRIPTION
**Pull request by manuelvanrijn**
I've come across an issue where the `identifier` wasn't equal to the `audience` in the token. This resulted in verification errors because currently it will verify the `aud` against the `identifier` if no `audience` is specified.

In this PR, I introduced the `audience` as `client_options` and will pass this along in the `verify!` of the `decoded_id_token` so the openid_connect gem [can handle the expected audience](https://github.com/nov/openid_connect/blob/e1eb8ea962af43752b1aed2c1063a3e24f96c5bc/lib/openid_connect/response_object/id_token.rb#L30-L32)

### CODE REVIEW

**Comment by manuelvanrijn** [comment]
@stanhu just a friendly ping 😄 Would you have time to review this ? Thanks in advance!

File: lib/omniauth/strategies/openid_connect.rb [hunk 1]
```diff
@@ -466,6 +467,7 @@ def verify_id_token!(id_token)
 
         decode_id_token(id_token).verify!(issuer: options.issuer,
                                           client_id: client_options.identifier,
+                                          audience: client_options.audience,
```
Comment (stanhu) [inline lib/omniauth/strategies/openid_connect.rb:470]: Would it be safer to pass in `audience` only if it is present? I noticed in https://github.com/nov/openid_connect/blob/a3ec9145dcb637e420ebbb9b1858bdadfbe3e575/lib/openid_connect/response_object/id_token.rb#L30 that the logic might behave differently if `audience` were a blank string, for exmaple.

Comment (manuelvanrijn) [reply lib/omniauth/strategies/openid_connect.rb:470]: Good catch. Seems safer to only pass it if present.

I'll update it and let you know

File: lib/omniauth/strategies/openid_connect.rb [hunk 2]
```diff
@@ -464,9 +465,14 @@ def configured_response_type
...
 
-        decode_id_token(id_token).verify!(issuer: options.issuer,
-                                          client_id: client_options.identifier,
-                                          nonce: params['nonce'].presence || stored_nonce)
+        verify_kwargs = {
+          issuer: options.issuer,
+          client_id: client_options.identifier,
+          nonce: params['nonce'].presence || stored_nonce
+        }
+        verify_kwargs.merge!(audience: client_options.audience) if client_options.audience.present?
```
Comment (stanhu) [inline lib/omniauth/strategies/openid_connect.rb:473]: Just a nit:

```suggestion
        verify_kwargs[:audience] = client_options.audience if client_options.audience.present?
```

File: lib/omniauth/strategies/openid_connect.rb [hunk 3]
```diff
@@ -464,9 +465,14 @@ def configured_response_type
...
 
-        decode_id_token(id_token).verify!(issuer: options.issuer,
-                                          client_id: client_options.identifier,
-                                          nonce: params['nonce'].presence || stored_nonce)
+        verify_kwargs = {
+          issuer: options.issuer,
+          client_id: client_options.identifier,
+          nonce: params['nonce'].presence || stored_nonce
+        }
+        verify_kwargs[:audience] = client_options.audience if client_options.audience.present?
```
Comment (stanhu) [inline lib/omniauth/strategies/openid_connect.rb:473]: I think `present?` is only available with ActiveSupport:

```suggestion
        verify_kwargs[:audience] = client_options.audience unless client_options.audience.to_s.empty?
```

Comment (stanhu) [reply lib/omniauth/strategies/openid_connect.rb:473]: Or maybe it suffices to do this:

```suggestion
        verify_kwargs[:audience] = client_options.audience if client_options.audience
```

**Comment by stanhu** [comment]
There are also Rubocop failures: https://github.com/omniauth/omniauth_openid_connect/actions/runs/9779345570/job/27001862614?pr=179

File: lib/omniauth/strategies/openid_connect.rb (hunk 3 above)
Comment (manuelvanrijn) [reply lib/omniauth/strategies/openid_connect.rb]: done!

Comment (stanhu) [reply lib/omniauth/strategies/openid_connect.rb]: I'm surprised Rubocop didn't complain about the use of `merge!` for a single item, but 🤷

**Review by stanhu — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -108,6 +108,7 @@ These are the configuration options for the client_options hash of the configura
 | scheme                 | The http scheme to use                                          | https      |                        |
 | host                   | The host of the authorization server                            | nil        |                        |
 | port                   | The port for the authorization server                           | 443        |                        |
+| audience               | The intended consumer of the token                              | nil        |                        |
```
Comment (bufferoverflow) [inline README.md:111]:
```suggestion
| audience               | The intended consumer (`aud` field) of the id_token              | nil        |                        |
```

**Review by bufferoverflow — APPROVED** [review]
Looks good, just one little docs suggestion. Feel free to ignore it.

**Pull request merged**